trio/_core/_multierror.py

```diff
@@ -186,9 +193,7 @@ def __new__(cls, exceptions):
         if len(exceptions) == 1:
             return exceptions[0]
         else:
-            self = BaseException.__new__(cls)
-            self.exceptions = exceptions
-            return self
+            return BaseException.__new__(cls, exceptions)
```

Comment:
Well.... okay, but this is super confusing and it still took me a good 5 minutes to figure out what was happening even in the context of the discussion above. So how about some more comments:

```python
if len(exceptions) == 1:
    # If this object happens to itself be a MultiError, then Python will
    # implicitly call our __init__ again, which we don't want and have
    # guard against.
    return exceptions[0]
else:
    # This always implicitly invokes __init__, which is what we want.
    return BaseException.__new__(cls, exceptions)
```